docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,403 @@
+
+```bash
+cargo fmt --all -- --config group_imports=StdExternalCrate
+```
+
+### Import order
+Code files should have a maximum of three import groups:
+1. A group with std/alloc/core imports
+2. A group with external crate imports
+3. A group for internal imports: self, super, and crate.
+
+This grouping and ordering is enforced by the rustfmt flag, `group_imports`.
+
+### Code block consistency
+
+Surrounding code blocks with `{` and `}` is encouraged, even when the enclosed
+block is a single statement.  Blocks in the same lexical scope must use
+consistent conventions.  For example, consider the following:
+
+```
+match foo {
+1..2 => {
+// this is a single statement, but it is surrounded
+// with { and } because the other blocks in the match
+// statement need them.
+Ok(true)
+},
+3..4 => {
+error!("Bad value for foo");
+Err(Error::BadFoo)
+},
+_ => {
+// similarly, this block uses { }
+Ok(true)
+}
+}
+
+// conversely, all of the arms of this match statement
+// have one-statement blocks, so { and } can be elided.
+match bar {
+1..2 => Some("abc"),
+3..4 => Some("def"),
+_ => None
+}
+```
+
+### Whitespace
+
+All contributions should use the same whitespacing as the rest of the project.
+Moreover, Pull requests where a large number of changes only deal with whitespace will be
+rejected.
+
+## Comments
+Comments are very important for the readability and correctness of the codebase. The purpose of comments is:
+
+* Allow readers to understand the roles of components and functions without having to check how they are used.
+* Allow readers to check the correctness of the code against the comments.
+* Allow readers to follow tests.
+
+In the limit, if there are no comments, the problems that arise are:
+
+* Understanding one part of the code requires understanding *many* parts of the code. This is because the reader is forced to learn the meanings of constructs inductively through their use. Learning how one construct is used requires understanding its neighbors, and then their neighbors, and so on, recursively. Instead, with a good comment, the reader can understand the role of a construct with `O(1)` work by reading the comment.
+* The user cannot be certain if there is a bug in the code, because there is no distinction between the contract of a function, and its definition.
+* The user cannot be sure if a test is correct, because the logic of the test is not specified, and the functions do not have contracts.
+
+### Comment Formatting
+
+Comments are to be formatted in typical `rust` style, specifically:
+
+- Use markdown to format comments.
+
+- Use the triple forward slash "///" for modules, structs, enums, traits and functions. Use double forward slash "//" for comments on individual lines of code.
+
+- Start with a high-level description of the function, adding more sentences with details if necessary.
+
+- When documenting panics, errors, or other conceptual sections, introduce a Markdown section with a single `#`, e.g.:
+
+- ```
```

Comment:
There is an extraneous `-` here which messes up the formatting for the rest of the [doc](https://github.com/stacks-network/stacks-blockchain/blob/972f061bb5a8dc0ccb200c648d71286a7eabf066/docs/CONTRIBUTING.md).

Reply:
the indent had to be fixed here!